bin/sce_to_anndata.R

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env Rscript
+
+# This script takes a SingleCellExperiment stored in a .rds file and converts it
+# to an AnnData object saved as an hdf5 file
+
+# import libraries
+suppressPackageStartupMessages({
+  library(optparse)
+  library(SingleCellExperiment)
+})
+# set up arguments
+option_list <- list(
+  make_option(
+    opt_str = c("-i", "--input_sce_file"),
+    type = "character",
+    help = "path to rds file with input sce object to be converted"
+  ),
+  make_option(
+    opt_str = c("-o", "--output_h5_file"),
+    type = "character",
+    help = "path to output hdf5 file to store AnnData object. Must end in .hdf5 or .h5"
+  )
+)
+
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# check that filtered SCE file exists
+if(!file.exists(opt$input_sce_file)){
+  stop(glue::glue("{opt$input_sce_file} does not exist."))
+}
+
+# check that output file is h5
+if(!(stringr::str_ends(opt$output_h5_file, ".hdf5|.h5"))){
+  stop("output file name must end in .hdf5 or .h5")
+}
+
+# Convert to AnnData -----------------------------------------------------------
+
+# read in sce
+sce <- readr::read_rds(opt$input_sce_file)
+
+# export sce as anndata object
+scpcaTools::sce_to_anndata(sce,
+                           anndata_file = opt$output_h5_file)
```

Comment:
This is incredibly minor, but follows from my feeling that we may want to standardize on tidyverse code formatting. 
```suggestion
scpcaTools::sce_to_anndata(
  sce,
  anndata_file = opt$output_h5_file
)
```

I'm also just wondering how this function does handling ADT data? Do we need to convert that separately if it is present? (I have not looked, but I'm kind of assuming you did when writing this)

Reply:
> Yes, so this does not handle alternative experiments. Our function that we are using will only convert the main experiment. Looking back on [AlexsLemonade/scpcaTools#115](https://github.com/AlexsLemonade/scpcaTools/issues/115), we had some initial thoughts on how we wanted to handle it, one of which was outputting a separate file for each `altExp`. I think we may want to address this in a separate issue/PR here because we will have to think about what we want the output to look like there.
> 
> In looking briefly at the Scanpy documentation, it looks like they store everything in one matrix with the adt data as additional rows in the gene by cell counts matrix. Maybe we could do something similar prior to converting to anndata to keep everything in one file? https://scanpy-tutorials.readthedocs.io/en/multiomics/cite-seq/pbmc5k.html

Doing that seems kind of hacky, and I don't love it. I think maybe the right approach going forward is to to export `mudata` objects? https://mudata.readthedocs.io/en/latest/. This allows wrapping multiple anndata objects in a way much more similar to SCE. The accessing the underlying AnnData objects is done with calls like `mudata['rna']`. 

There are some remaining questions though: For example: do we want all files to be `mudata` for output, even if there is only RNA data?

All of this I think falls into future discussion, but we should probably resolve it pretty soon to prevent too much rewriting later.

Reply:
I'm going to file a new issue about this, but think we should tackle it after this goes in/ this sprint. Another question I have is about making sure our output is compliant with CZI cellxgene, since that is part of the goal of creating the `AnnData` output. I think because of that we may have to keep everything as `AnnData` rather than use `mudata`? 
https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.0.0/schema.md#general-requirements

That being said, it sounds like they are starting to figure out CITE-seq, so maybe we could discuss with them, how best to store the CITE-seq data.

Reply:
Agree on a new issue. But I think that we can do both pretty easily, as we should be able to make the AnnData RNA object within the `mudata` compatible with cellxgene.